File: src/main/java/ru/reosfire/wmlib/yaml/common/TimeConfig.java
package ru.reosfire.wmlib.yaml.common;

import org.bukkit.configuration.ConfigurationSection;
import ru.reosfire.wmlib.yaml.YamlConfig;

public class TimeConfig extends YamlConfig
{
    public final long Seconds;
    public TimeConfig(ConfigurationSection configurationSection)
    {
        super(configurationSection);
        long seconds = configurationSection.getInt("Seconds", 0);
        seconds += configurationSection.getLong("Minutes", 0) * 60;
        seconds += configurationSection.getLong("Hours", 0) * 60 * 60;
        seconds += configurationSection.getLong("Days", 0) * 60 * 60 * 24;

        Seconds = seconds;
    }
}

File: src/main/java/ru/reosfire/wmlib/yaml/common/TimeConfig.kt
package ru.reosfire.wmlib.yaml.common

import org.bukkit.configuration.ConfigurationSection
import ru.reosfire.wmlib.yaml.YamlConfig

class TimeConfig(configurationSection: ConfigurationSection) : YamlConfig(configurationSection) {
    val milliseconds: Long

    init {
        var ms = getLong("Milliseconds", 0)
        ms += getLong("Seconds", 0) * 1000
        ms += getLong("Minutes", 0) * 1000 * 60
        ms += getLong("Hours", 0)   * 1000 * 60 * 60
        ms += getLong("Days", 0)    * 1000 * 60 * 60 * 24
        milliseconds = ms
    }
}